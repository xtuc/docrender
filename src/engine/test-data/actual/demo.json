{
  "title": "Publish-Bot demo",
  "body": {
    "content": [
      {
        "endIndex": 1,
        "sectionBreak": {
          "sectionStyle": {
            "columnSeparatorStyle": "NONE",
            "contentDirection": "LEFT_TO_RIGHT",
            "sectionType": "CONTINUOUS"
          }
        }
      },
      {
        "startIndex": 1,
        "endIndex": 19,
        "paragraph": {
          "elements": [
            {
              "startIndex": 1,
              "endIndex": 19,
              "textRun": {
                "content": "Turn any document\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.vleoh07olg87",
            "namedStyleType": "TITLE",
            "alignment": "CENTER",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 19,
        "endIndex": 34,
        "paragraph": {
          "elements": [
            {
              "startIndex": 19,
              "endIndex": 34,
              "textRun": {
                "content": "into a website\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.vleoh07olg87",
            "namedStyleType": "TITLE",
            "alignment": "CENTER",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 34,
        "endIndex": 35,
        "paragraph": {
          "elements": [
            {
              "startIndex": 34,
              "endIndex": 35,
              "textRun": {
                "content": "\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "alignment": "CENTER",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 35,
        "endIndex": 95,
        "paragraph": {
          "elements": [
            {
              "startIndex": 35,
              "endIndex": 95,
              "textRun": {
                "content": "Publish Bot allows you to turn any document into a website.\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.w99vcdbqjel7",
            "namedStyleType": "HEADING_2",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 95,
        "endIndex": 162,
        "paragraph": {
          "elements": [
            {
              "startIndex": 95,
              "endIndex": 111,
              "textRun": {
                "content": "This is a demo. ",
                "textStyle": {}
              }
            },
            {
              "startIndex": 111,
              "endIndex": 112,
              "textRun": {
                "content": "M",
                "textStyle": {
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 1
                      }
                    }
                  }
                }
              }
            },
            {
              "startIndex": 112,
              "endIndex": 113,
              "textRun": {
                "content": "o",
                "textStyle": {
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 1,
                        "green": 0.6
                      }
                    }
                  }
                }
              }
            },
            {
              "startIndex": 113,
              "endIndex": 114,
              "textRun": {
                "content": "s",
                "textStyle": {
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 0.6,
                        "blue": 1
                      }
                    }
                  }
                }
              }
            },
            {
              "startIndex": 114,
              "endIndex": 115,
              "textRun": {
                "content": "t",
                "textStyle": {
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 0.15294118,
                        "green": 0.30588236,
                        "blue": 0.07450981
                      }
                    }
                  }
                }
              }
            },
            {
              "startIndex": 115,
              "endIndex": 135,
              "textRun": {
                "content": " of the Google Docs ",
                "textStyle": {}
              }
            },
            {
              "startIndex": 135,
              "endIndex": 143,
              "textRun": {
                "content": "features",
                "textStyle": {
                  "italic": true
                }
              }
            },
            {
              "startIndex": 143,
              "endIndex": 148,
              "textRun": {
                "content": " are ",
                "textStyle": {}
              }
            },
            {
              "startIndex": 148,
              "endIndex": 157,
              "textRun": {
                "content": "supported",
                "textStyle": {
                  "bold": true
                }
              }
            },
            {
              "startIndex": 157,
              "endIndex": 162,
              "textRun": {
                "content": " üòÄ.\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 162,
        "endIndex": 163,
        "paragraph": {
          "elements": [
            {
              "startIndex": 162,
              "endIndex": 163,
              "textRun": {
                "content": "\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 163,
        "endIndex": 180,
        "paragraph": {
          "elements": [
            {
              "startIndex": 163,
              "endIndex": 180,
              "textRun": {
                "content": "And code blocks:\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 180,
        "endIndex": 186,
        "paragraph": {
          "elements": [
            {
              "startIndex": 180,
              "endIndex": 186,
              "textRun": {
                "content": "```js\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 186,
        "endIndex": 214,
        "paragraph": {
          "elements": [
            {
              "startIndex": 186,
              "endIndex": 214,
              "textRun": {
                "content": "console.log(‚Äúcode example‚Äù)\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 214,
        "endIndex": 218,
        "paragraph": {
          "elements": [
            {
              "startIndex": 214,
              "endIndex": 218,
              "textRun": {
                "content": "```\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 218,
        "endIndex": 220,
        "paragraph": {
          "elements": [
            {
              "startIndex": 218,
              "endIndex": 219,
              "inlineObjectElement": {
                "inlineObjectId": "kix.uiranxs7r93n",
                "textStyle": {}
              }
            },
            {
              "startIndex": 219,
              "endIndex": 220,
              "textRun": {
                "content": "\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "alignment": "CENTER",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 220,
        "endIndex": 221,
        "paragraph": {
          "elements": [
            {
              "startIndex": 220,
              "endIndex": 221,
              "textRun": {
                "content": "\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "alignment": "CENTER",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 221,
        "endIndex": 346,
        "paragraph": {
          "elements": [
            {
              "startIndex": 221,
              "endIndex": 249,
              "textRun": {
                "content": "See the source Google Docs: ",
                "textStyle": {}
              }
            },
            {
              "startIndex": 249,
              "endIndex": 345,
              "textRun": {
                "content": "https://docs.google.com/document/d/1nSn4UBI-xo0tE5PwJOq7Dp3TgLlhT7b-dM3rK2xcHsk/edit?usp=sharing",
                "textStyle": {
                  "underline": true,
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 0.06666667,
                        "green": 0.33333334,
                        "blue": 0.8
                      }
                    }
                  },
                  "link": {
                    "url": "https://docs.google.com/document/d/1nSn4UBI-xo0tE5PwJOq7Dp3TgLlhT7b-dM3rK2xcHsk/edit?usp=sharing"
                  }
                }
              }
            },
            {
              "startIndex": 345,
              "endIndex": 346,
              "textRun": {
                "content": "\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      }
    ]
  },
  "documentStyle": {
    "background": {
      "color": {}
    },
    "pageNumberStart": 1,
    "marginTop": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginBottom": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginRight": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginLeft": {
      "magnitude": 72,
      "unit": "PT"
    },
    "pageSize": {
      "height": {
        "magnitude": 841.8897637795277,
        "unit": "PT"
      },
      "width": {
        "magnitude": 595.27559055118115,
        "unit": "PT"
      }
    },
    "marginHeader": {
      "magnitude": 36,
      "unit": "PT"
    },
    "marginFooter": {
      "magnitude": 36,
      "unit": "PT"
    },
    "useCustomHeaderFooterMargins": true
  },
  "namedStyles": {
    "styles": [
      {
        "namedStyleType": "NORMAL_TEXT",
        "textStyle": {
          "bold": false,
          "italic": false,
          "underline": false,
          "strikethrough": false,
          "smallCaps": false,
          "backgroundColor": {},
          "foregroundColor": {
            "color": {
              "rgbColor": {}
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          },
          "weightedFontFamily": {
            "fontFamily": "Arial",
            "weight": 400
          },
          "baselineOffset": "NONE"
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "alignment": "START",
          "lineSpacing": 115,
          "direction": "LEFT_TO_RIGHT",
          "spacingMode": "COLLAPSE_LISTS",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "unit": "PT"
          },
          "borderBetween": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderTop": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderBottom": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderLeft": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderRight": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "indentFirstLine": {
            "unit": "PT"
          },
          "indentStart": {
            "unit": "PT"
          },
          "indentEnd": {
            "unit": "PT"
          },
          "keepLinesTogether": false,
          "keepWithNext": false,
          "avoidWidowAndOrphan": true,
          "shading": {
            "backgroundColor": {}
          },
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_1",
        "textStyle": {
          "fontSize": {
            "magnitude": 20,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 20,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 6,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_2",
        "textStyle": {
          "bold": false,
          "fontSize": {
            "magnitude": 16,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 18,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 6,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_3",
        "textStyle": {
          "bold": false,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.2627451,
                "green": 0.2627451,
                "blue": 0.2627451
              }
            }
          },
          "fontSize": {
            "magnitude": 14,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 16,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_4",
        "textStyle": {
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 12,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 14,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_5",
        "textStyle": {
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 12,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_6",
        "textStyle": {
          "italic": true,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 12,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "TITLE",
        "textStyle": {
          "fontSize": {
            "magnitude": 26,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 3,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "SUBTITLE",
        "textStyle": {
          "italic": false,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 15,
            "unit": "PT"
          },
          "weightedFontFamily": {
            "fontFamily": "Arial",
            "weight": 400
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 16,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      }
    ]
  },
  "revisionId": "ALBJ4LtmDz5htvoWmoASuyvIofa5fMpnDmSMvFkej6Q_OGkGl9lIcAblmmuzQGc7oODywG31w6fSnRPgGxct3U0",
  "suggestionsViewMode": "SUGGESTIONS_INLINE",
  "inlineObjects": {
    "kix.uiranxs7r93n": {
      "objectId": "kix.uiranxs7r93n",
      "inlineObjectProperties": {
        "embeddedObject": {
          "imageProperties": {
            "contentUri": "https://lh7-rt.googleusercontent.com/docsz/AD_4nXf39uRj45OCD9jV4sAWeI17BZbc57elpnSdLcgNUffYYzOTAUrlgWJCOjDfkk8IylG9qhc-kDMCG86Q7XtZA9FCRjhScN5WmQzplq2V_avUbO7xZBUDQXc3ynuF9znmyItfIARqwIoJlO4B-KITBcy6XLE?key=pCWSwNGfNMAcn7cyLin2Lg",
            "cropProperties": {}
          },
          "embeddedObjectBorder": {
            "color": {
              "color": {
                "rgbColor": {}
              }
            },
            "width": {
              "unit": "PT"
            },
            "dashStyle": "SOLID",
            "propertyState": "NOT_RENDERED"
          },
          "size": {
            "height": {
              "magnitude": 67.139040064844821,
              "unit": "PT"
            },
            "width": {
              "magnitude": 66.487204724409423,
              "unit": "PT"
            }
          },
          "marginTop": {
            "magnitude": 9,
            "unit": "PT"
          },
          "marginBottom": {
            "magnitude": 9,
            "unit": "PT"
          },
          "marginRight": {
            "magnitude": 9,
            "unit": "PT"
          },
          "marginLeft": {
            "magnitude": 9,
            "unit": "PT"
          }
        }
      }
    }
  },
  "documentId": "1nSn4UBI-xo0tE5PwJOq7Dp3TgLlhT7b-dM3rK2xcHsk"
}
