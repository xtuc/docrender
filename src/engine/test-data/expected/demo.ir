(
    content: [
        Clear,
        Title((
            level: 1,
            text: Text("Turn any document"),
        ), [
            ("text-align", "center"),
            ("margin-left", "auto"),
            ("margin-right", "auto"),
        ]),
        Clear,
        Title((
            level: 1,
            text: Text("into a website"),
        ), [
            ("text-align", "center"),
            ("margin-left", "auto"),
            ("margin-right", "auto"),
        ]),
        Clear,
        Paragraph([]),
        Title((
            level: 3,
            text: Text("Publish Bot allows you to turn any document into a website."),
        ), []),
        Clear,
        Paragraph([
            InlineText([
                Text("This is a demo. "),
            ], false),
            InlineText([
                TextWithOptions("M", (
                    styles: [
                        ("color", "rgb(100%, 0%, 0%)"),
                    ],
                )),
            ], false),
            InlineText([
                TextWithOptions("o", (
                    styles: [
                        ("color", "rgb(100%, 60.000004%, 0%)"),
                    ],
                )),
            ], false),
            InlineText([
                TextWithOptions("s", (
                    styles: [
                        ("color", "rgb(60.000004%, 0%, 100%)"),
                    ],
                )),
            ], false),
            InlineText([
                TextWithOptions("t", (
                    styles: [
                        ("color", "rgb(15.294118%, 30.588236%, 7.4509807%)"),
                    ],
                )),
            ], false),
            InlineText([
                Text(" of the Google Docs "),
            ], false),
            InlineText([
                TextWithOptions("features", (
                    styles: [
                        ("font-style", "italic"),
                    ],
                )),
            ], false),
            InlineText([
                Text(" are "),
            ], false),
            InlineText([
                TextWithOptions("supported", (
                    styles: [
                        ("font-weight", "bold"),
                    ],
                )),
            ], false),
            InlineText([
                Text(" üòÄ."),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                Text("And code blocks:"),
            ], true),
        ]),
        Clear,
        Code("js", "console.log(‚Äúcode example‚Äù)\n"),
        Clear,
        Paragraph([
            Image("https://lh7-rt.googleusercontent.com/docsz/AD_4nXf39uRj45OCD9jV4sAWeI17BZbc57elpnSdLcgNUffYYzOTAUrlgWJCOjDfkk8IylG9qhc-kDMCG86Q7XtZA9FCRjhScN5WmQzplq2V_avUbO7xZBUDQXc3ynuF9znmyItfIARqwIoJlO4B-KITBcy6XLE?key=pCWSwNGfNMAcn7cyLin2Lg", ("66", "67"), [
                ("text-align", "center"),
                ("margin-left", "auto"),
                ("margin-right", "auto"),
            ]),
        ]),
        Clear,
        Paragraph([]),
        Paragraph([
            InlineText([
                Text("See the source Google Docs: "),
            ], false),
            InlineText([
                TextWithOptions("https://docs.google.com/document/d/1nSn4UBI-xo0tE5PwJOq7Dp3TgLlhT7b-dM3rK2xcHsk/edit?usp=sharing", (
                    link: Some("https://docs.google.com/document/d/1nSn4UBI-xo0tE5PwJOq7Dp3TgLlhT7b-dM3rK2xcHsk/edit?usp=sharing"),
                    styles: [
                        ("text-decoration", "underline"),
                        ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                    ],
                )),
            ], false),
        ]),
        Paragraph([]),
    ],
    styles: {
        "pbNORMAL_TEXT": [
            ("line-height", "1.7249999"),
        ],
    },
)