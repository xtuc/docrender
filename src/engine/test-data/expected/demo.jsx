import { createElement, Fragment } from "https://unpkg.com/es-react";

function Heading({ style, id, level, children }) {
  const tag = "h" + level;
  return createElement(tag, { style, id }, children);
}

function Paragraph({ children }) {
  return createElement("p", {}, children);
}

function Code({ children }) {
  return createElement("code", {}, children);
}

function Image({ src }) {
  return createElement("img", { src });
}

function Link({ href }) {
  return createElement("a", { href });
}

function List({ children }) {
  return createElement("ul", {}, children);
}

function ListItem({ children }) {
  return createElement("li", {}, children);
}

export default function Doc({
  heading = Heading,
  paragraph = Paragraph,
  code = Code,
  image = Image,
  link = Link,
  list = List,
  listItem = ListItem,
} = {}) {
  return createElement(
    Fragment,
    {},
    createElement(
      heading,
      {
        level: 1,
        style: { textAlign: "center", marginLeft: "auto", marginRight: "auto" },
        id: "pb-todo",
      },
      `Turn any document`,
    ),
    createElement(
      heading,
      {
        level: 1,
        style: { textAlign: "center", marginLeft: "auto", marginRight: "auto" },
        id: "pb-todo",
      },
      `into a website`,
    ),
    createElement(paragraph, {}),
    createElement(
      heading,
      { level: 3, id: "pb-todo" },
      `Publish Bot allows you to turn any document into a website.`,
    ),
    createElement(
      paragraph,
      {},
      `This is a demo. `,
      createElement("span", { style: { color: "rgb(100%, 0%, 0%)" } }, `M`),
      createElement(
        "span",
        { style: { color: "rgb(100%, 60.000004%, 0%)" } },
        `o`,
      ),
      createElement(
        "span",
        { style: { color: "rgb(60.000004%, 0%, 100%)" } },
        `s`,
      ),
      createElement(
        "span",
        { style: { color: "rgb(15.294118%, 30.588236%, 7.4509807%)" } },
        `t`,
      ),
      ` of the Google Docs `,
      createElement("span", { style: { fontStyle: "italic" } }, `features`),
      ` are `,
      createElement("span", { style: { fontWeight: "bold" } }, `supported`),
      ` üòÄ.`,
      `
`,
    ),
    createElement(
      paragraph,
      {},
      `And code blocks:`,
      `
`,
    ),
    createElement(
      code,
      { class: "language-js" },
      `<codeblock-js>console.log(‚Äúcode example‚Äù)
</codeblock-js>`,
    ),
    createElement(
      paragraph,
      {},
      createElement(image, {
        style: { textAlign: "center", marginLeft: "auto", marginRight: "auto" },
        src: "ed9dc5e3980bcdbcb51a29458192ca155592e122affef1bf7bf11e9e7012bce3",
        height: "67",
        width: "66",
      }),
    ),
    createElement(paragraph, {}),
    createElement(
      paragraph,
      {},
      `See the source Google Docs: `,
      createElement(
        link,
        {
          style: {
            textDecoration: "underline",
            color: "rgb(6.666667%, 33.333336%, 80%)",
          },
          href: "https://docs.google.com/document/d/1nSn4UBI-xo0tE5PwJOq7Dp3TgLlhT7b-dM3rK2xcHsk/edit?usp=sharing",
        },
        `https://docs.google.com/document/d/1nSn4UBI-xo0tE5PwJOq7Dp3TgLlhT7b-dM3rK2xcHsk/edit?usp=sharing`,
      ),
    ),
    createElement(paragraph, {}),
  );
}
