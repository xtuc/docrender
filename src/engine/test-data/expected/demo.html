<style>
  .pbNORMAL_TEXT {
    line-height: 1.7249999;
  }
</style>
<h1
  style="text-align: center; margin-left: auto; margin-right: auto"
  id="pb-Turn%20any%20document"
>
  Turn any document
</h1>
<h1
  style="text-align: center; margin-left: auto; margin-right: auto"
  id="pb-into%20a%20website"
>
  into a website
</h1>
<h3
  id="pb-Publish%20Bot%20allows%20you%20to%20turn%20any%20document%20into%20a%20website."
>
  Publish Bot allows you to turn any document into a website.
</h3>
<p>
  This is a demo. <span style="color: rgb(100%, 0%, 0%)">M</span
  ><span style="color: rgb(100%, 60.000004%, 0%)">o</span
  ><span style="color: rgb(60.000004%, 0%, 100%)">s</span
  ><span style="color: rgb(15.294118%, 30.588236%, 7.4509807%)">t</span> of the
  Google Docs <span style="font-style: italic">features</span> are
  <span style="font-weight: bold">supported</span> üòÄ.
</p>
<p>And code blocks:</p>
<pre><code class="language-js"><codeblock-js>console.log(‚Äúcode example‚Äù)
</codeblock-js></code></pre>
<style>
  script[visible],
  style[visible],
  pre {
    display: block;
    white-space: pre;
    border: 1px solid #dedede;
    padding: 1em;
    background: #fafafa;
    overflow-x: auto;
    border-left: 0.4em solid cornflowerblue;
    tab-size: 2;
    color: #1a1a1a;
    line-height: 1.6;
  }

  code:not(pre code), /* output:not(code:has(output) output) */ {
    background: #f7f7f7;
    border: 1px solid rgb(0 0 0 / 0.2);
    padding: 0.1rem 0.3rem;
    margin: 0.1rem 0;
    border-radius: 0.2rem;
    display: inline-block;
  }
</style>
<p>
  <img
    style="text-align: center; margin-left: auto; margin-right: auto"
    src="ed9dc5e3980bcdbcb51a29458192ca155592e122affef1bf7bf11e9e7012bce3"
    height="67"
    width="66"
  />
</p>
<p>
  See the source Google Docs:
  <a
    style="text-decoration: underline; color: rgb(6.666667%, 33.333336%, 80%)"
    href="https://docs.google.com/document/d/1nSn4UBI-xo0tE5PwJOq7Dp3TgLlhT7b-dM3rK2xcHsk/edit?usp=sharing"
    >https://docs.google.com/document/d/1nSn4UBI-xo0tE5PwJOq7Dp3TgLlhT7b-dM3rK2xcHsk/edit?usp=sharing</a
  >
</p>
